src/main/java/com/aws/greengrass/util/platforms/windows/WindowsExec.java

```diff
@@ -118,6 +116,88 @@ public synchronized void close() throws IOException {
             return;
         }
 
+        try {
+            stopGracefully();
+        } finally {
+            stopForcefully();
+        }
+    }
+
+    @SuppressFBWarnings("SWL_SLEEP_WITH_LOCK_HELD")
+    private void stopGracefully() {
+        int pid = ((ProcessImplForWin32) process).getPid();
+        boolean sentConsoleCtrlEvent = false;
+        synchronized (Kernel32.INSTANCE) {
+            // First, start a separate process that holds the console alive
+            // so that later gg can re-attach to this same console
+            Process holderProc;
+            try {
+                // Waits indefinitely for a keystroke
+                holderProc = new ProcessBuilder().command("cmd", "/C", "pause").start();
+            } catch (IOException e) {
+                logger.atError(STOP_GRACEFULLY_EVENT).cause(e).log("Failed to stop gracefully");
+                return;
+            }
+
+            Kernel32 k32 = Kernel32.INSTANCE;
+            try {
+                // Must detach from current console before attaching to another
+                if (!k32.FreeConsole()) {
+                    logger.atError(STOP_GRACEFULLY_EVENT).log("FreeConsole error {}", k32.GetLastError());
+                }
+                // Attach to the console that's running the target process
+                if (!k32.AttachConsole(pid)) {
+                    logger.atError(STOP_GRACEFULLY_EVENT).log("AttachConsole error {}", k32.GetLastError());
+                    return;
+                }
+                // Make gg ignore Ctrl-C temporarily so that we don't terminate ourselves as well
+                if (!Kernel32Ex.INSTANCE.SetConsoleCtrlHandler(null, true)) {
+                    logger.atError(STOP_GRACEFULLY_EVENT)
+                            .log("SetConsoleCtrlHandler add error {}", k32.GetLastError());
+                    return;
+                }
+                // Send Ctrl-C to all processes in the console
+                if (k32.GenerateConsoleCtrlEvent(Wincon.CTRL_C_EVENT, 0)) {
+                    sentConsoleCtrlEvent = true;
+                    // Wait to ensure CtrlHandler is not enabled before the calling process gets the signal
+                    try {
+                        Thread.sleep(2000);
+                    } catch (InterruptedException ignore) {
+                        logger.atWarn(STOP_GRACEFULLY_EVENT).log("CtrlHandler sleep interrupted");
+                        Thread.interrupted();  // clear the interrupted status
```

Comment:
This is not correct. `Thread.currentThread().interrupt()` is what we want; we need to re-interrupt the thread.
`interrupted()` returns you a boolean, it doesn't do anything

Reply:
`interrupted` does clear the interrupted status according to https://docs.oracle.com/javase/8/docs/api/java/lang/Thread.html#interrupted--

Reply:
We do not want to clear the interrupt flag though. Ignoring the interrupt is not right

Reply:
Got it I misunderstood "reset"

Reply:
Yes sorry, re-set as in set the flag again